File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    start, end = 1, 1
    sum_ = 1
    
    while start <= n:
        if n > sum_:
            end += 1
            sum_ += end
        elif n < sum_:
            sum_ -= start
            start += 1
        else:
            answer += 1
            sum_ -= start
            start += 1
    
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        int left = 0;
        int right = 0;
        int sum = 0;
        
        while (left <= right && right <= n) {
            if (sum <= n) {
                right++;
                sum += right;
            } else if (sum > n) {
                sum -= left;
                left++;
            }
            if (sum == n) {
                answer++;
            }
        }
        return answer;
    }
}